src/box/sql/build.c

```diff
@@ -115,6 +118,24 @@ encode_format(struct region *region, struct space_def *def, uint32_t *size)
 			mpstream_encode_str(&stream, "default");
 			mpstream_encode_str(&stream, default_str);
 		}
+		if (field->constraint_count == 0)
+			continue;
+		mpstream_encode_str(&stream, "foreign_key");
+		mpstream_encode_map(&stream, field->constraint_count);
+		for (uint32_t i = 0; i < field->constraint_count; ++i) {
+			struct tuple_constraint_def *cdef =
+				&field->constraint_def[i];
+			mpstream_encode_str(&stream, cdef->name);
```

Comment:
I have a non-related question - aren't `CHECK`s also tuple constraints? Are they stored in box as `CONSTR_FUNC`? I suppose for their encoding in SQL we don't use `tuple_constraint_def` yet, do we?

Reply:
True, my mistake was not to support them from the very beginning. Fixed in another PR.